### PR TITLE
Smart Hose Timer Support

### PR DESCRIPTION
**Pull request by brg468**
Adding the new host and endpoints for the smart hose timer. Let me know what I screwed up 😛. 

I had to remove Python 3.6 to get the tests to run, not sure if 3.10-11 should be added. Also this is my first major stab at tests so they might not be totally correct.

### CODE REVIEW

**Comment by brg468** [comment]
I'm not sure what needs to be done since 3.6 says its required but fails, I'm gonna hold off making any more changes for now. 

Also not sure what hound is complaining about, would be nice if it said what Black had issues with. If I let Black run it fails the linters with too long lines.

**Comment by rfverbruggen** [comment]
Thanks for your PR @brg468. I'll have a look at it soon. Together we'll get to a working solution!

**Comment by rfverbruggen** [comment]
> I'm not sure what needs to be done since 3.6 says its required but fails, I'm gonna hold off making any more changes for now.

I've updated the dev branch (and this one as well) with the latest supported [Python versions](https://devguide.python.org/versions/), so that's covered for now.

File: rachiopy/rachioobject.py [hunk 1]
```diff
@@ -100,3 +101,74 @@ def delete_request(self, path: str, body=None):
         :rtype: tuple
         """
         return self._request(path, "DELETE", body)
+
+    def _valve_request(self, path: str, method: str, body=None):
+        """Make a request from the API.
```
Comment (rfverbruggen) [inline rachiopy/rachioobject.py:106]:
```suggestion
        """Make a request to the API.
```

I see that the `_request` method also has this wrong. Could you improve that one as well?

File: tests/test_valve.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        self.assertEqual(args[1], f"{VALVE_API_URL}/valve/setDefaultRuntime")
+        self.assertEqual(args[0], "PUT")
+        self.assertEqual(
+            kwargs["data"],
+            json.dumps(
+                {"valveId": valveid, "defaultRuntimeSeconds": duration}
+            ),
+        )
+
+        # Check that values should be within range.
```
Comment (rfverbruggen) [inline tests/test_valve.py:113]: I would make these separate test methods to explicitly have a 'happy flow' test and 'exception tests'. Could you process that?

File: tests/test_valve.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+        # Check that the mock function is called with the rights args.
+        self.assertEqual(args[1], f"{VALVE_API_URL}/valve/startWatering")
+        self.assertEqual(args[0], "PUT")
+        self.assertEqual(
+            kwargs["data"],
+            json.dumps({"valveId": valveid, "durationSeconds": duration}),
+        )
+
+        # Check that values should be within range.
```
Comment (rfverbruggen) [inline tests/test_valve.py:141]: For this one, the same comment applies as above. Could you process that?

**Review by rfverbruggen — CHANGES_REQUESTED** [review]

**Comment by rfverbruggen** [comment]
@all-contributors please add @brg468  for code, doc, test

File: tests/test_valve.py (hunk 2 above)
Comment (brg468) [reply tests/test_valve.py:122]: Let me know if that's what you meant, happy to fix if not.

**Review by rfverbruggen — APPROVED** [review]

**Comment by rfverbruggen** [comment]
Thanks for the effort @brg468. Will make an official release ASAP.

**Comment by rfverbruggen** [comment]
@brg468 it's now published as version 1.1.0. If you have any improvements later on please submit them and I'm happy to help integrate them into the Rachiopy package.

**Comment by brg468** [comment]
I saw you bumped the version in HA. Do you have an update to integrate the hose timer in HA? If not I’ve been working on it for a bit and I think it’s about ready to submit over there. Happy to collaborate if you also have something working.

**Comment by rfverbruggen** [comment]
Haven't started working on it. Feel free to create your own PR, tag me and I'm happy to review and/or contribute.

**Pull request merged**